pdns/pdns_recursor.cc

```diff
@@ -959,7 +959,7 @@ void startDoResolve(void *p)
           }
 
           // Does the query or validation mode sending out a SERVFAIL on validation errors?
-          if(!pw.getHeader()->cd && (g_dnssecmode == DNSSECMode::ValidateAll || (dc->d_mdp.d_header.ad && g_dnssecmode != DNSSECMode::Off))) {
+          if(!pw.getHeader()->cd && (g_dnssecmode == DNSSECMode::ValidateAll || (dc->d_mdp.d_header.ad && (g_dnssecmode != DNSSECMode::Off || g_dnssecmode != DNSSECMode::ProcessNoValidate)))) {
```

Comment:
I think we can drop the whole `&& (g_dnssecmode != DNSSECMode::Off || g_dnssecmode != DNSSECMode::ProcessNoValidate)` as indeed the earlier condition prevents this